### ISSUE TITLE
Deprecation warning

### ISSUE THREAD

**Issue by hseeberger** [scala/pickling#299]
"method newTermName in trait Names is deprecated: Use TermName instead"

**Comment by phaller** [comment]
Yep, it would be nice to resolve this. The problem is that we haven't figured out a way to do that without creating a 2.11-specific (or 2.10-specific) branch. Medium-term there's probably no way around it, though.

**Comment by Morgaroth** [comment]
How about version specific source directory? like in [@ExNexu project](https://github.com/ExNexu/spray-json-annotation/tree/master/src/main)
Sorry if it is stupid question xD

**Comment by the21st** [comment]
Our project uses `-Xfatal-warnings` (which is common best practice) so this a deal-breaker for us.

**Comment by yukw777** [comment]
+1

**Comment by jvican** [comment]
We'll try to work around this soon.

**Comment by the21st** [comment]
@jvican any progress on this?

**Comment by jvican** [comment]
No, but feel free to contribute :smile:.
